tools/publish_aws.py

```diff
@@ -90,14 +90,15 @@ def __init__(
             self._artifact_paths.append(artifact_path)
 
 
-    def _upload_artifact(self, filepath):
+    def _upload_artifact(self, filepath, content_type=None):
         filename = os.path.basename(filepath)
+        cmd = 'aws s3'
         if self._aws_region:
-            cmd = 'aws s3 --region={} cp --acl public-read {} {}/{} 1>&2'.format(
-                self._aws_region, filepath, self._s3_directory, filename)
-        else:
-            cmd = 'aws s3 cp --acl public-read {} {}/{} 1>&2'.format(
-                filepath, self._s3_directory, filename)
+            cmd += ' --region={}'.format(self._aws_region)
+        cmd += ' cp --acl public-read'
+        if content_type:
+            cmd += ' --content-type "{}"'.format(content_type)
+        cmd += ' {} {}/{} 1>&2'.format(filepath, self._s3_directory, filename)
```

Comment:
Not really specific to this code, but I strongly prefer command-as-vector over command-as-big-string  Programs that break when a dir introduces a space is so 1980.